pgjdbc/src/main/java/org/postgresql/core/UTF8Encoding.java

```diff
@@ -15,10 +15,11 @@
   private static final int MIN_4_BYTES = 0x10000;
   private static final int MAX_CODE_POINT = 0x10ffff;
 
-  private char[] decoderArray = new char[1024];
+  private char[] decoderArray;
 
   UTF8Encoding(String jvmEncoding) {
     super(jvmEncoding);
+    decoderArray = new char[Integer.parseInt(System.getProperty("PG_UTF8_ENC_BYTES", String.valueOf(1024)))];
```

Comment:
@atuldaemon , did you just add this "just in case"?
Typically we use connection properties for settings.